src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -344,7 +344,7 @@ private void validateSchema(final EventTypeBase eventType) throws InvalidEventTy
             }
 
             if (eventType.getCompatibilityMode() != CompatibilityMode.NONE) {
-                validateJsonSchemaConstraints(schemaAsJson);
+                validateJsonSchemaConstraints(eventType, schemaAsJson);
```

Comment:
Here it seems that only the compatibility mode is used to validate the constraints, rather than the whole event type. Would it make sense to pass the compatibility mode only?